lib/srv/alpnproxy/conn_upgrade.go

```diff
@@ -75,6 +82,51 @@ func IsALPNConnUpgradeRequired(addr string, insecure bool) bool {
 	return result
 }
 
+func isRemoteNoALPNError(err error) bool {
+	var opErr *net.OpError
+	return errors.As(err, &opErr) && opErr.Op == "remote error" && strings.Contains(opErr.Err.Error(), "tls: no application protocol")
+}
+
+// OverwriteALPNConnUpgradeRequirementByEnv overwrites ALPN connection upgrade
+// requirement by environment variable.
+func OverwriteALPNConnUpgradeRequirementByEnv(addr string) (bool, bool) {
+	envValue := os.Getenv(defaults.TLSRoutingConnUpgradeEnvVar)
+	if envValue == "" {
+		return false, false
+	}
+	result := isALPNConnUpgradeRequiredByEnv(addr, envValue)
+	logrus.WithField(defaults.TLSRoutingConnUpgradeEnvVar, envValue).Debugf("ALPN connection upgrade required for %q: %v.", addr, result)
+	return result, true
+}
+
```

Comment:
```suggestion

// isALPNConnUpgradeRequiredByEnv parses the given environment
// variable value as a series of values separated by either "," or ";".
// Settings for a specific address can be given with"<address>=<value>".
// Example:
// TELEPORT_TLS_ROUTING_CONN_UPGRADE="yes,example.teleport.com:443=no".
// This will require ALPN connection upgrade for all addresses
// except "example.teleport.com:443".
```
nit: It was easier to understand this function after I read the test cases for it, so a docstring would be helpful here, or I guess just saying like "this parses TELEPORT_TLS_ROUTING_CONN_UPGRADE, see docs for example settings".